### DISCUSSION TITLE
Allow users to set their timezone and locale

### DISCUSSION THREAD

**Discussion by luchaos** [RetroAchievements/RAWeb#704] [Ideas]
For displaying timestamps in web interface. Leverage Carbon PHP library features for transformation on backend side and moment.js for the frontend.

Related:

#701

**Comment by KingS1zzle** [comment]
This would be a cool quality of life feature. Since I joined I decided to add a second clock just to keep track of when things are posted. 

This isn't that much of a hassle but would be a better solution than adding a second clock to your pc and hovering over to see when it was posted.
![image](https://user-images.githubusercontent.com/64910342/186041295-a8f7d778-da28-4873-a4dc-7a956e49cee0.png)

**Reply by redwizard42** [reply]
Yeah. I always see the timestamps on the website and my brain goes, "What's that in local time?" and then I forget what timezone the site is in and I give up.

**Reply by suspect15** [reply]
I hate to admit it, but I'm the same way.  GMT basically means nothing to me as I can never easily orient myself to it despite it only being a simple subtraction.

**Comment by Xeinok-Pals** [comment]
This would still definitely be a huge upgrade in terms of maturity and QoL - are there any major technical hurdles to this?

**Reply by wescopeland** [reply]
Yes. This has about the same complexity as username changes until the UI is migrated to React.

**Comment by thatonedood** [comment]
Commenting here just to show support for this idea, would be awesome to be implemented. I understand that there are technical hurdles, just voicing my support.

**Comment by vizthex123** [comment]
i'd love to have this. My dumb brain assumes it's just a simple "subtract X hours from the tooltip when users hover over it", but i'm sure it's a lot more complicated than that lol.

**Comment by CosmicLime** [comment]
Now that would be a very useful feature for RA, it's like people said above, there are users who struggle very much with time zones including myself, so having a system like that would be a huge painkiller. I know there are lots of barrier in implementing these things, but please I ask that you guys don't abandon this idea!

**Reply by vizthex123** [reply]
yeah, it could just be a client-side conversion rather than having to change the timezone the site uses (which i'm pretty sure is UTC? So the site could go "oh, you're 4 hours behind UTC so we'll display that as the time" but it's still tracked in UTC on the backend).

i dunno if that would be easier than a full conversion or anything, but the result for the end user is the same - so i'd saying using the easier solution would be better.
